pkg/preparer/orchestrate.go

```diff
@@ -212,6 +206,14 @@ func (p *Preparer) verifySignature(manifest pods.Manifest, logger logging.Logger
 func (p *Preparer) installAndLaunchPod(newManifest *pods.Manifest, pod Pod, logger logging.Logger) bool {
 	// do not remove the logger argument, it's not the same as p.Logger
 
+	// HACK ZONE. When we have better authz, rewrite.
+	if ((newManifest.RunAsUser() == "root") && (newManifest.ID() != POD_ID)) || p.podIdForbidden(newManifest.RunAsUser()) {
```

Comment:
I don't think this boolean logic is correct. the `||` means it will fail the deploy of the preparer, even if it's the preparer ID.